### PR TITLE
Update try-it document

### PR DESCRIPTION
**Pull request by fmuyassarov**
Preview at: https://deploy-preview-130--metal3.netlify.com/try-it.html

### CODE REVIEW

File: _includes/try-it.md [hunk 1]
```diff
@@ -1,3 +1,276 @@
...
+machine.cluster.x-k8s.io "test1-controlplane-0" deleted
+baremetalmachine.infrastructure.cluster.x-k8s.io "test1-controlplane-0" deleted
+kubeadmconfig.bootstrap.cluster.x-k8s.io "test1-controlplane-0" deleted
+
+# deprovision cluster
+$ ./scripts/v1alpha2/delete_cluster.sh
+cluster.cluster.x-k8s.io "test1" deleted
+baremetalcluster.infrastructure.cluster.x-k8s.io "test1" deleted
+```
+After running above scripts you should see that deprovisioning started
```
Comment (Xenwar) [inline _includes/try-it.md]: deprovisioning has started

File: _includes/try-it.md [hunk 2]
```diff
@@ -1,3 +1,276 @@
...
+kube-system   kube-controller-manager-node-1             1/1     Running   0          120m
+kube-system   kube-proxy-9tmvb                           1/1     Running   0          120m
+kube-system   kube-proxy-nfln4                           1/1     Running   0          81m
+kube-system   kube-scheduler-node-1                      1/1     Running   0          120m
+```
+
+Deprovision Cluster and Machines
+------------
+
+You can run the following scripts to deprovision your target cluster and
```
Comment (Xenwar) [inline _includes/try-it.md]: the target cluster

File: _includes/try-it.md [hunk 3]
```diff
@@ -1,3 +1,276 @@
...
+
+<p align="center">
+  <img width="564" height="534" src="../metal3-dev-env.svg">
+</p>
+
+Prerequisites
+------------
+The minimum requirements for the machine hosting Metal³-dev-env can be
+different depending on the Linux distribution of the target node. The following
+table shows supported distributions and their requirements.
```
Comment (Xenwar) [inline _includes/try-it.md]: their resource requirements

Comment (fmuyassarov) [reply _includes/try-it.md]: all resolved

**Comment by alosadagrande** [comment]
@fmuyassarov  IMHO it is not clear the differences between the new content added (which is great) and the old one, which starts with "Detailed step-by-step instructions". Is it planned to be replaced by the new one?

Actually the old content it is not included in the index either.

**Comment by fmuyassarov** [comment]
> @fmuyassarov IMHO it is not clear the differences between the new content added (which is great) and the old one, which starts with "Detailed step-by-step instructions". Is it planned to be replaced by the new one?
> 
> Actually the old content it is not included in the index either.

Yes, I agree that there is not much of differences, except some updates on variables and having high-level architecture & demo. It is not planned to replace the current content and I'm waiting for reviews, based on that it will be maybe decided to either update the current content by replacing it or add missing parts on top of the current one.

**Comment by kashifest** [comment]
Excellent work @fmuyassarov , would it be possible to do a merge of the content with the previous one and in case there are duplicated information, remove it. I can see there are two prerequisite section now for example.

File: _includes/try-it.md [hunk 4]
```diff
@@ -1,3 +1,286 @@
...
+* [Before you begin](#before-you-begin)
+   * [Architecture](#architecture)
+   * [Prerequisites](#prerequisites)
+   * [Environment variables](#environment-variables)
+* [Quick start](#quick-start)
+   * [Deploying Metal3-dev-env](#deploy-metal3-dev-env)
+   * [Provision Cluster and Machines](#provision-cluster-and-machines)
+   * [Deprovision Cluster and Machines](#deprovision-cluster-and-machines)
+   * [Tear down Metal3-dev-env](#tear-down-metal3-dev-env)
+
```
Comment (maelk) [inline _includes/try-it.md]: What about adding the detailed step by step instructions there too ? To remove the duplicated TOC

File: _includes/try-it.md [hunk 5]
```diff
@@ -1,25 +1,48 @@
...
 
-## Instructions
+# Instructions
 
 ### Prerequisites
 
  * System with CentOS 7 or Ubuntu 18.04
  * Bare metal preferred, as we will be creating VMs to emulate bare metal hosts
  * Run as a user with passwordless sudo access
+ * Resource requirements for the host machine vary depending on the selected
```
Comment (Xenwar) [inline _includes/try-it.md]: for the host machines

File: _includes/try-it.md [hunk 6]
```diff
@@ -34,20 +57,32 @@ The `Makefile` runs a series of scripts, described here:
 * `04_verify.sh` - Runs a set of tests that verify that the deployment completed
```
Comment (Xenwar) [inline _includes/try-it.md]: the deployment has completed successfully.

File: _includes/try-it.md [hunk 7]
```diff
@@ -34,20 +57,32 @@ The `Makefile` runs a series of scripts, described here:
...
-To tear down the environment, run `make clean`.
+To tear down the environment, run
+
+```sh
+$ make clean
+```
 
-You can also run some tests provisioning and deprovisioning machines by running
-`make test`
+**Note**: you can also run some tests provisioning and deprovisioning machines
```
Comment (Xenwar) [inline _includes/try-it.md]: some tests of

File: _includes/try-it.md [hunk 8]
```diff
@@ -106,128 +145,164 @@ used to create these host objects is in `bmhosts_crs.yaml`.
...
-NAME       STATUS    PROVISIONING STATUS   MACHINE   BMC                         HARDWARE PROFILE   ONLINE    ERROR
-node-0     OK        ready                           ipmi://192.168.111.1:6230   unknown            true
-node-1     OK        ready                           ipmi://192.168.111.1:6231   unknown            true
+NAME     STATUS   PROVISIONING STATUS   CONSUMER   BMC                         HARDWARE PROFILE   ONLINE   ERROR
+node-0   OK       ready                            ipmi://192.168.111.1:6230   unknown            true     
+node-1   OK       ready                            ipmi://192.168.111.1:6231   unknown            true     
 ```
 
 You can also look at the details of a host, including the hardware information
 gathered by doing pre-deployment introspection.
```
Comment (Xenwar) [inline _includes/try-it.md]: gathered by the pre-deployment introspection.

File: _includes/try-it.md [hunk 9]
```diff
@@ -106,128 +145,164 @@ used to create these host objects is in `bmhosts_crs.yaml`.
...
+09:10:38.926489       controller-runtime/controller "msg"="Starting workers"  "controller"="baremetalmachine" "worker count"=1
+10:54:16.943712       Host matched hostSelector for BareMetalMachine
+10:54:16.943772       2 hosts available while choosing host for bare metal machine
+10:54:16.944087       Associating machine with host
+10:54:17.516274       Finished creating machine
+10:54:17.518718       Provisioning BaremetalHost
+
 ```
 
 If you look at the yaml representation of the `Machine`, you will see a new
```
Comment (Xenwar) [inline _includes/try-it.md]: `Machine`'s request

File: _includes/try-it.md [hunk 10]
```diff
@@ -249,99 +324,58 @@ provisioned and associated with a `Machine`.
 ```sh
 $ kubectl get baremetalhosts -n metal3
 
-NAME       STATUS    PROVISIONING STATUS   MACHINE   BMC                         HARDWARE PROFILE   ONLINE    ERROR
-node-0     OK        ready                           ipmi://192.168.111.1:6230   unknown            true
-node-1     OK        provisioning          centos    ipmi://192.168.111.1:6231   unknown            true
+NAME     STATUS   PROVISIONING STATUS   CONSUMER               BMC                         HARDWARE PROFILE   ONLINE   ERROR
```
Comment (Xenwar) [inline _includes/try-it.md]: comment for line 321.

that one of the host is now

File: _includes/try-it.md [hunk 11]
```diff
@@ -249,99 +324,58 @@ provisioned and associated with a `Machine`.
...
+$ kubectl get baremetalhosts -n metal3
+NAME     STATUS   PROVISIONING STATUS   CONSUMER               BMC                         HARDWARE PROFILE   ONLINE   ERROR
+node-0   OK       deprovisioning        test1-md-0-m87bq       ipmi://192.168.111.1:6230   unknown            false     
+node-1   OK       deprovisioning        test1-controlplane-0   ipmi://192.168.111.1:6231   unknown            false
 
-```sh
-./01_prepare_host.sh
-./02_configure_host.sh
-./03_launch_mgmt_cluster.sh
+$ kubectl get cluster -n metal3
```
Comment (Xenwar) [inline _includes/try-it.md]: Add a text above this to introduce the cluster (probably why it is also deprovisioning)

File: _includes/try-it.md [hunk 12]
```diff
@@ -425,15 +445,14 @@ You will then see the host go into a `deprovisioning` status:
...
 ```
 
-## Running a Custom baremetal-operator
+## Running Custom Baremetal-Operator
 
 The `baremetal-operator` comes up running in the cluster by default, using an
-image built from the `metal3-io/baremetal-operator` github repository.  If
-you’d like to test changes to the `baremetal-operator`, you can follow this
-process.
+image built from the [metal3-io/baremetal-operator](https://github.com/metal3-io/baremetal-operator) repository.  If you’d like to test changes to the
```
Comment (Xenwar) [inline _includes/try-it.md]: you’d like to monitor changes to the ?

File: _includes/try-it.md [hunk 13]
```diff
@@ -468,7 +489,7 @@ cd ~/go/src/github.com/metal3-io/cluster-api-provider-baremetal
 make run
 ```
 
-## Accessing the Ironic API
+## Accessing Ironic API
 
 Sometimes you may want to look directly at Ironic to debug something.
```
Comment (Xenwar) [inline _includes/try-it.md]: Sometimes you may want to look directly at Ironic to debug an issue.

**Comment by fmuyassarov** [comment]
@russellb can you please have a look at this PR?

**Comment by iranzo** [comment]
Seems that this PR requires a rebase as it conflicts with the try-it.md document, can you rebase @fmuyassarov

**Comment by fmuyassarov** [comment]
@iranzo rebased and updated

**Comment by alosadagrande** [comment]
@fmuyassarov I am missing some note when provisioning CentOS 7 as the OS. I think it is needed to create at least this export explicitly or in the config_x.sh

export IMAGE_NAME_CENTOS=centos-update.qcow2

I would add the lookup in the Ansible role as you suggested:

IMAGE_NAME_CENTOS: "{{ lookup('env', 'IMAGE_NAME_CENTOS') | default('centos-update.qcow2.', true)}}"

**Comment by fmuyassarov** [comment]
> @fmuyassarov I am missing some note when provisioning CentOS 7 as the OS. I think it is needed to create at least this export explicitly or in the config_x.sh
> 
> export IMAGE_NAME_CENTOS=centos-update.qcow2
> 
> I would add the lookup in the Ansible role as you suggested:
> 
> IMAGE_NAME_CENTOS: "{{ lookup('env', 'IMAGE_NAME_CENTOS') | default('centos-update.qcow2.', true)}}"

Thanks.
Yes, the lookup is missing there with a default value as you have posted here.
But, I think the default value still should be "CentOS-7-x86_64-GenericCloud-1907.qcow2" based on [/images.sh](https://github.com/metal3-io/metal3-dev-env/blob/master/lib/images.sh)
What do do you think?

**Comment by alosadagrande** [comment]
My point here is, where is used the official image: CentOS-7-x86_64-GenericCloud-1907.qcow2 ?

At least in v1alpha2 I am always running the specific centos-update.qcow2 image.

**Comment by fmuyassarov** [comment]
> My point here is, where is used the official image: CentOS-7-x86_64-GenericCloud-1907.qcow2 ?
> 
> At least in v1alpha2 I am always running the specific centos-update.qcow2 image.
> 
> Let me know if I am missing something, but in that case members should always have to change the defaults if they want to run v1alpha2 and probably v1alpha3, right? Why not setting the defaults to something that works out of the box?
> 
> The default cloud image for CentOS will be changed once we get to CentOS 8 or we are planning to maintain both?

@alosadagrande 
sorry, I was wrong, indeed we need the updated one.
Regarding the CentOS8, IMO I don't think that we are planning to maintain both CentOS7/CentOS8.
I think we will drop the CentOS7 once we move to CentOS8.

**Comment by fmuyassarov** [comment]
@alosadagrande 
Personally, I haven't provisioned CentOS based target machines, that's why I didn't modify the variable and didn't know that centos-updated is the only one that could work. But thanks for raising that issue.

**Comment by alosadagrande** [comment]
@fmuyassarov you think we can merge it or you prefer to wait for the changes you are making to the Ansible provisioning scripts (https://github.com/metal3-io/metal3-dev-env/pull/229)

**Comment by fmuyassarov** [comment]
> @fmuyassarov you think we can merge it or you prefer to update the PR with the changes you are doing to the Ansible provisioning scripts ([metal3-io/metal3-dev-env#229](https://github.com/metal3-io/metal3-dev-env/pull/229))

I think we can merge now, and once [PR#229](https://github.com/metal3-io/metal3-dev-env/pull/229) goes in, we can update the doc as well respectively.

**Comment by maelk** [comment]
Let's proceed with the merge then. @russellb could you please have a look and approve if ok for you ?

**Pull request merged**